registry/storage/tagstore_test.go

```diff
@@ -205,5 +219,75 @@ func TestTagLookup(t *testing.T) {
 	if len(tags) != 2 {
 		t.Errorf("Lookup of descB returned %d tags, expected 2", len(tags))
 	}
+}
+
+func TestTagIndexes(t *testing.T) {
+	env := testTagStore(t)
+	tagStore := env.ts
+	ctx := env.ctx
+
+	indexes, ok := tagStore.(distribution.TagIndexes)
+	if !ok {
+		t.Fatal("tagStore does not implement TagIndexes interface")
+	}
+
+	conf, err := env.bs.Put(ctx, "application/octet-stream", []byte{0})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	dgstsSet := make(map[digest.Digest]bool)
+	for i := 0; i < 3; i++ {
```

Comment:
Maybe make this an array of cases with different tags, I think we should make sure that other tags don't also end up in the list.